cmds/ip/ip.go

```diff
@@ -258,13 +329,12 @@ func main() {
 	}()
 
 	// The ip command doesn't actually follow the BNF it prints on error.
-	// There are lots of handy shortcuts that people will expect.
 	switch arg[cursor] {
-	case "addr":
+	case "address", "addres", "addre", "addr", "add", "ad", "a":
```

Comment:
yeah we definitely need a function for this :-)

Reply:
https://play.golang.org/p/Y1MM7rcvfW
might do. You call it with the arg[cursor], the minimum you will take, and the rest you must match.

in this case
match(arg[cursor], "a", "ddress") 
would work
you can change it to also work like this
match(arg[cursor], "a", "address")

Reply:
I agree it will be better